language/move-vm/runtime/src/loader.rs

```diff
@@ -618,13 +645,7 @@ impl Loader {
         data_store: &mut dyn DataStore,
     ) -> VMResult<CompiledModule> {
         let module = match data_store.load_module(id) {
```

Comment:
yeah that is the part I was questioning with the comment I had. I would love for this code to stay here rather than relying on the data cache to pass a valid `CompiledModule`